### ISSUE TITLE
enforce-slice-style: Add declaration style

### ISSUE THREAD

**Issue by jalaziz** [mgechev/revive#971]
**Is your feature request related to a problem? Please describe.**

In https://github.com/mgechev/revive/issues/898, it was mentioned:
> There is also a way of having a non-initialized slice (which should be preferred over the empty slice w/o len and cap in most cases), but this is beyond the scope of this linter

This preferred way is the var declaration style:
```
var emptySlice []int
```

This is the preferred style by both the Golang and Uber style guides:
* https://google.github.io/styleguide/go/decisions#nil-slices
* https://github.com/uber-go/guide/blob/master/style.md#local-variable-declarations

**Describe the solution you'd like**

It would be great if the `enforce-slice-style` rule could be configured to prefer the var declaration style. The tricky part may be that this style is only preferred when declaring a new variable, but other styles may be preferred when assigning an existing variable or passing a parameter.

**Describe alternatives you've considered**

The only alternative I have considered is disabling the rule.

**Additional context**

**Comment by denisvmedia** [comment]
I guess it should be easy to add: it's basically erroring on both: `literal` AND `make` style initialization. Are you willing to submit a PR? 😉

**Comment by jalaziz** [comment]
> Are you willing to submit a PR? 😉

I'm definitely willing to try. I'm not terribly familiar with Golang linting, but willing to learn. Likely can't get to this until the weekend or maybe next weekend, but will try to find time to work on it.

**Comment by denisvmedia** [comment]
> I think it might be more than this because you can't use var style declarations when assigning to a struct member or passing a literal to a function (both of which revive will warn about today).

Well, you don't need to assign them or you can assign nil.

**Comment by jalaziz** [comment]
> Well, you don't need to assign them if it's a brand new instance of the struct or you can assign nil.

Ah yeah, fair point. For function argument literals it can still be nice for readability, but certainly shouldn't be needed for structs (although, it could help towards the JSON serialization issue).

Will see what I can figure out when I take a deeper look.

**Comment by denisvmedia** [comment]
> For function argument literals it can still be nice for readability

Even for the function arguments you can do it this way:

```go
func dummy(v []int) {
}

dummy([]int(nil))
```

This would be consistent with the rule you propose.

**Comment by jalaziz** [comment]
> This would be consistent with the rule extension you propose.

Right again. I clearly need sleep 😂. Thanks for the advice!
